elastic/fleet-server
# Alter checkin API to remove attributes set by audit/unenroll API

## What is the problem this PR solves?

Follow up to https://github.com/elastic/fleet-server/pull/3818, alters the checkin API to remove attributes that the audit/unenroll endpoint sets

## How does this PR solve the problem?

Alter the checkin wrapper's flush method to include bulk update calls to remove the attributes for agents that have a non-empty `audit_unenrolled_reason`.

## How to test this PR locally

1. Enroll an agent
2. Use the `audit/unenroll` API to annotate the agent document.
3. Use the `checkin` api to signal the agent is healthy

Agent document in Elasticsearch should no longer have the `audit_unenrolled_reason` or associated attributes

## Design Checklist


## Checklist

- ~~I have made corresponding changes to the documentation~~
- ~~I have made corresponding change to the default configuration files~~

## Related issues

- Relates https://github.com/elastic/elastic-agent/issues/484
